File: javaProblems/linkedlist/compare_ll.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
    public int solve(ListNode A, ListNode B) {
        ListNode h1 = A;
        ListNode h2 = B;
        while (h1 != null || h2 != null){
            if (h1.val != h2.val){
                return 0;
            }
            h1 = h1.next;
            h2 = h2.next;
        }
        if (h1 != null || h2 != null){
            return 0;
        }
        return 1;
    }
}

File: pyProblems/linkedlist/compare_ll.py
# Definition for singly-linked list.
# class ListNode:
#    def __init__(self, x):
#        self.val = x
#        self.next = None

class Solution:
    # @param A : head node of linked list
    # @param B : head node of linked list
    # @return an integer
    def solve(self, A, B):
        h1 = A
        h2 = B
        while h1 is not None or h2 is not None:
            if h1.val != h2.val:
                return 0
            h1 = h1.next
            h2 = h2.next
        if h1 is not None or h2 is not None:
            return 0
        return 1
